### PR TITLE
37 implement rqspline using distrax

### PR DESCRIPTION
**Pull request by kazewong**
Implementation is done. 

The compilation time of the part seems to be pretty significant (One minute for training loop and one minute for sampling). Need to knock that down

### CODE REVIEW

**Comment by kazewong** [comment]
The RQspline NF model can have runaway behavior after long training, which probably has to do with points going outside the RQ spline region. I think some normalization layers should fix that.

**Comment by kazewong** [comment]
Smaller learning rate seems to alleviate the behaviour, tho we should still take a close look to the issue

**Comment by kazewong** [comment]
@marylou-gabrie I have merged the branch, but I will leave this PR open so you can have a look. Once you have finish checking the branch, please delete the branch.
